File: atcoder/atcoder_problems/agc/agc001~100/agc027/agc027_a.py
N, X = map(int, input().split())
A = list(map(int, input().split()))

A.sort()

ans = 0
if X > sum(A):
    ans -= 1


for a in A:
    if X >= a:
        X -= a
    else:
        break
    ans += 1
    



print(ans)

File: atcoder/atcoder_problems/agc/agc001~100/agc027/agc027_a.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <numeric>
using namespace std;

int main(){
    int N, X;
    cin >> N >> X;
    vector<int> A(N);
    for (int i = 0; i < N; i++) cin >> A[i];

    sort(A.begin(), A.end());
    long long A_sum = accumulate(A.begin(), A.end(),0LL);

    if (A_sum == X){
        cout << N << endl;
        return 0;
    }

    
    int ans = 0;
    for (int i = 0; i < N; i++){
        if (X < A[i]) break;
        X -= A[i];
        ans += 1;
    }
    if (ans == N) ans -= 1;
    cout << ans << endl;
    return 0;
}